woocommerce/woocommerce
# Product task experiment updates

### Changes proposed in this Pull Request:

Follow-up to https://github.com/woocommerce/woocommerce/pull/32941.

- Rename functions to be more consistent
- Move out experiment codes to onboarding package so it can be used by the package
- Add ExPlat call for PHP part of product task experiment
- Add experiment check for product import
- Add experiment check for trackView

### How to test the changes in this Pull Request:

**Prerequisite**
1. Install `WooCommerce Admin Test Helper`
2. Go to `Tools > WCA Test Helper > Features`
3. Enable `experimental-products-task`
4. Make sure your site has `woocommerce_allow_tracking` option set to `yes`

**Testing old import if not treatment**
1. Install `WooCommerce Admin Test Helper`
4. Go to `Tools > WCA Test Helper > Experiments`
2. Disable both `woocommerce_products_task_layout_stacked` and `woocommerce_products_task_layout_card` for both frontend and backend experiment
3. Make sure there are no products in store (if there are in trash, empty them)
1. Go to appearance task `/wp-admin/admin.php?page=wc-admin&task=appearance`
2. Click on "Import products" button
4. Go to Products page
6. Observe 18 products are imported

**Testing import under treatment**
1. Enable treatment for either `woocommerce_products_task_layout_stacked` or `woocommerce_products_task_layout_card` **for backend and frontend**
1. Trash all products & empty trash
2. Go to Products task
3. Click on "Load sample products"
4. Observe 9 products are imported

**Testing trackView**
1. Enable treatment for either `woocommerce_products_task_layout_stacked` or `woocommerce_products_task_layout_card` **for frontend**
2. Open up developer console and make sure tracks are debugged `localStorage.setItem( 'debug', 'wc-admin:*' );`
3. Go to Products task
4. Observe `wcadmin_task_view` track in developer console with `experimental_products: true` prop

### Other information:




### FOR PR REVIEWER ONLY:
